lib/MusicBrainz/Server/WebService/XMLSerializer.pm

```diff
@@ -281,19 +283,27 @@ sub _serialize_release_group
             Interview => 5
         );
 
-        my ($fallback) =
+        my ($fallback_name) =
             nsort_by { $fallback_type_order{$_} }
                 grep { exists $fallback_type_order{$_} }
                     map { $_->name }
                         $release_group->all_secondary_types;
+        my ($fallback_gid) =
+            nsort_by { $fallback_type_order{$_} }
+                grep { exists $fallback_type_order{$_} }
```

Comment:
`%fallback_type_order` doesn't have gids for keys, so nothing will exist there. You can try removing the `map { $_->name }` above, use `$_->name` as the key into `%fallback_type_order`, and then do `$fallback->name` and `$fallback->gid` below.